src/sqlfluff/core/file_helpers.py

```diff
@@ -0,0 +1,15 @@
+"""File Helpers for the parser module."""
+
+import chardet
+
+from sqlfluff.core.config import FluffConfig
+
+
+def get_encoding(fname: str, config: FluffConfig) -> str:
+    """Get the encoding of the file (autodetect)."""
+    encoding_config = config.get("encoding", default="autodetect")
+
+    if encoding_config == "autodetect":
+        return chardet.detect(open(fname, "rb").read())["encoding"]
```

Comment:
Let's make sure to close the file. While CPython will automatically close the file when the reference goes out of scope, other versions may not (e.g. [PyPy](https://www.pypy.org/)).